Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/Batching/Batching.xaml.cs

```diff
@@ -0,0 +1,240 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Windows.Storage;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Microsoft.AI.MachineLearning;
+
+namespace WinMLSamplesGallery.Samples
+{
+    public sealed class EvalResult
+    {
+        public string nonBatchedAvgTime { get; set; }
+        public string batchedAvgTime { get; set; }
+        public string timeRatio { get; set; }
+    }
+
+    public sealed partial class Batching : Page
+    {
+        private LearningModelSession nonBatchingSession_;
+        private LearningModelSession BatchingSession_;
+        float avgNonBatchedDuration = 0;
+        float avgBatchDuration = 0;
+        static bool navigatingAwayFromPage = false;
+
+        public Batching()
+        {
+            this.InitializeComponent();
+            // Ensure static variable is always false on page initialization
+            navigatingAwayFromPage = false;
+        }
+
+        async private void StartInference(object sender, RoutedEventArgs e)
+        {
+            ShowEvalUI();
+            ResetEvalMetrics();
+
+            var inputImages = GetInputImages();
+            int batchSize = GetBatchSizeFromBatchSizeComboBox();
+            CreateSessions(batchSize);
+
+            UpdateEvalText(false);
+            await Classify(inputImages);
+
+            UpdateEvalText(true);
+            await ClassifyBatched(inputImages, batchSize);
+
+            GenerateEvalResultAndUI();
+        }
+
+        private void ShowEvalUI()
+        {
+            StartInferenceBtn.IsEnabled = false;
+            EvalResults.Visibility = Visibility.Collapsed;
+            LoadingContainer.Visibility = Visibility.Visible;
+        }
+
+        private void ResetEvalMetrics()
+        {
+            avgNonBatchedDuration = 0;
+            avgBatchDuration = 0;
+        }
+
+        private List<VideoFrame> GetInputImages()
+        {
+            var birdFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/hummingbird.jpg")).GetAwaiter().GetResult();
+            var catFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/kitten.png")).GetAwaiter().GetResult();
+            var birdImage = CreateSoftwareBitmapFromStorageFile(birdFile);
+            var catImage = CreateSoftwareBitmapFromStorageFile(catFile);
+            var inputImages = new List<VideoFrame>();
+            for (int i = 0; i < 25; i++)
```

Comment:
why 25? also, please use a const variable instead so there's a name and intended usage associated with the number.